packages/typespec-test/test/batch_modular/generated/typespec-ts/src/models/pagingTypes.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * An interface that tracks the settings for paged iteration
+ */
+export interface PageSettings {
+  /**
+   * The token that keeps track of where to continue the iterator
+   */
+  continuationToken?: string;
+}
+
+/**
+ * An interface that describes a page of results.
+ */
+export type ContinuablePage<TElement, TPage = TElement[]> = TPage & {
+  /**
+   * The token that keeps track of where to continue the iterator
+   */
+  continuationToken?: string;
+};
+
+/**
+ * An interface that allows async iterable iteration both to completion and by page.
+ */
+export interface PagedAsyncIterableIterator<
+  TElement,
+  TPage = TElement[],
+  TPageSettings extends PageSettings = PageSettings
+> {
+  /**
+   * The next method, part of the iteration protocol
+   */
+  next(): Promise<IteratorResult<TElement>>;
+  /**
+   * The connection to the async iterator, part of the iteration protocol
+   */
+  [Symbol.asyncIterator](): PagedAsyncIterableIterator<
+    TElement,
+    TPage,
+    TPageSettings
+  >;
+  /**
+   * Return an AsyncIterableIterator that works a page at a time
+   */
+  byPage: (
+    settings?: TPageSettings
+  ) => AsyncIterableIterator<ContinuablePage<TElement, TPage>>;
+}
+
+/**
+ * An interface that describes how to communicate with the service.
+ */
+export interface PagedResult<
+  TElement,
+  TPage = TElement[],
+  TPageSettings extends PageSettings = PageSettings
+> {
+  /**
+   * Link to the first page of results.
+   */
+  firstPageLink?: string;
+  /**
+   * A method that returns a page of results.
+   */
+  getPage: (
+    pageLink?: string
+  ) => Promise<{ page: TPage; nextPageLink?: string } | undefined>;
+  /**
+   * a function to implement the `byPage` method on the paged async iterator.
+   */
+  byPage?: (
+    settings?: TPageSettings
+  ) => AsyncIterableIterator<ContinuablePage<TElement, TPage>>;
+
+  /**
+   * A function to extract elements from a page.
+   */
+  toElements?: (page: TPage) => unknown[];
```

Comment:
We initially typed this as `unknown` to preserve backward compatibility but this is not a concern in this PR so we can do the right thing and return TElement[].
```suggestion
  toElements?: (page: TPage) => TElement[];
```